flink-connectors/flink-connector-kafka-base/src/test/java/org/apache/flink/streaming/connectors/kafka/KafkaTestBase.java

```diff
@@ -218,6 +218,8 @@ protected void assertAtLeastOnceForTopic(
 		while (System.currentTimeMillis() < startMillis + timeoutMillis) {
 			properties.put("key.deserializer", "org.apache.kafka.common.serialization.IntegerDeserializer");
 			properties.put("value.deserializer", "org.apache.kafka.common.serialization.IntegerDeserializer");
+			properties.put("session.timeout.ms", "2000");
+			properties.put("heartbeat.interval.ms", "500");
```

Comment:
Might cause the tests to be unstable on travis (travis can easily stall for seconds and this decreases the timeout from 10s down to 2s).

Reply:
These configurations are only used when we are using Kafka based group management. Given that Flink does not use that in the UT when consuming from Kafka. These configs just need to be set so the configurations are "valid" to KafkaConsumer. And even if the consumers do hit timeout in UT, in the worst case a consumer will just rejoin the group and consume again. So it probably won't really hurt the test stability.

The reason we set these two configurations was that we reduced the request timeout to 3 seconds. And the `session.timeout.ms` and `heartbeat.interval.ms` cannot be longer than that in some old KafkaConsumer versions. I'll add a comment to explain that.

Reply:
Ok, thanks for the explanation. Coping/pasting this as a comment is also probably a good idea :)